cmd/kubeadm/app/phases/kubelet/config_unix.go

```diff
@@ -0,0 +1,51 @@
+// +build !windows
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubelet
+
+import (
+	"k8s.io/klog"
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
+	"k8s.io/kubernetes/pkg/util/procfs"
+	utilsexec "k8s.io/utils/exec"
+)
+
+// buildDynamicDefaults returns a Kubelet config file with corrected values according to different OS configuration
+func buildDynamicDefaults(kubeletConfig *kubeletconfig.KubeletConfiguration, nodeReg *kubeadmapi.NodeRegistrationOptions) *kubeletconfig.KubeletConfiguration {
```

Comment:
tests?

Reply:
will add it.

Reply:
@kad could you give some tips about writing test case for this private function? I tried, but still not figure out yet.

Reply:
@chenzhiwei please see the original tests of `buildKubeletArgMap`, but basically you need to supply some mocks in here (for `utilsexec` and `PidOf`) and then test with these.